atom/browser/api/atom_api_web_contents.cc

```diff
@@ -861,6 +861,12 @@ void WebContents::Observe(int type,
   }
 }
 
+void WebContents::BeforeUnloadDialogCancelled() {
+  if (deferred_load_url_.id) {
```

Comment:
Awesome, thanks for this feedback.  I figured there'd be some special case I hadn't considered.